public/app/plugins/datasource/loki/querybuilder/components/NestedQuery.test.tsx

```diff
@@ -0,0 +1,112 @@
+): NestedQueryProps => {
+  const nestedQuery: LokiVisualQueryBinary = {
+    operator: operator,
+    vectorMatchesType: vectorMatchesType,
+    query: {
+      labels: [],
+      operations: [],
+    },
+  };
+
+  const datasource = new LokiDatasource(
+    {
+      url: '',
+      jsonData: {},
+      meta: {} as DataSourcePluginMeta,
+    } as DataSourceInstanceSettings,
+    undefined,
+    undefined
+  );
+
+  const props: NestedQueryProps = {
+    nestedQuery: nestedQuery,
+    index: 0,
+    datasource: datasource,
+    onChange: jest.fn(),
+    onRemove: jest.fn(),
+    onRunQuery: jest.fn(),
+    showExplain: showExplain,
+  };
+
+  return props;
+};
+
+// All test assertions need to be awaited for, because the component uses `useEffect` to update the state.
+
+describe('render all elements', () => {
+  it('renders the operator label', async () => {
+    const props = createMockProps();
+    render(<NestedQuery {...props} />);
+    expect(await screen.findByText('Operator')).toBeInTheDocument();
+  });
+
+  it('renders the expected operator value', async () => {
+    const props = createMockProps('!=');
+    render(<NestedQuery {...props} />);
+    expect(await screen.findByText('!=')).toBeInTheDocument();
+  });
+
+  it('renders the vector matches label', async () => {
+    const props = createMockProps();
+    render(<NestedQuery {...props} />);
+    expect(await screen.findByText('Vector matches')).toBeInTheDocument();
+  });
+
+  it('renders the expected vector matches value', async () => {
+    const props = createMockProps(undefined, 'ignoring');
+    render(<NestedQuery {...props} />);
+    expect(await screen.findByText('ignoring')).toBeInTheDocument();
+  });
+});
+
+describe('exit the nested query', () => {
+  it('onRemove is called when clicking (x)', async () => {
+    const props = createMockProps();
+    render(<NestedQuery {...props} />);
+    fireEvent.click(await screen.findByLabelText('remove-nested-query'));
+    await waitFor(() => expect(props.onRemove).toHaveBeenCalledTimes(1));
+  });
+});
+
+describe('change operator', () => {
+  it('onChange is called with the correct args', async () => {
+    const props = createMockProps('/', 'on');
+    render(<NestedQuery {...props} />);
+    userEvent.click(await screen.findByLabelText('operator-select'));
+    fireEvent.click(await screen.findByText('+'));
+    await waitFor(() => expect(props.onChange).toHaveBeenCalledTimes(1));
+    await waitFor(() =>
+      expect(props.onChange).toHaveBeenCalledWith(0, {
+        operator: '+',
+        vectorMatchesType: 'on',
+        query: { labels: [], operations: [] },
+      })
+    );
+  });
+});
+
+describe('explain mode', () => {
+  it('shows the explanation when set to true', async () => {
+    const props = createMockProps(undefined, undefined, true);
+    render(<NestedQuery {...props} />);
+    expect(await screen.findByText('Fetch all log lines matching label filters.')).toBeInTheDocument();
+  });
+});
```

Comment:
I'd be really interested in adding coverage for this line, which will require interacting with `<LokiQueryBuilder>`:

https://github.com/grafana/grafana/blob/46e38402a1cb55c32632470bdd56c809ed8b6312/public/app/plugins/datasource/loki/querybuilder/components/NestedQuery.tsx#L84